### PR TITLE
Set a boundary between trusted and verified for zookeeper and fluent controller

### PR DESCRIPTION
**Pull request by euclidgame**

### CODE REVIEW

**Comment by euclidgame** [comment]
@marshtompsxd , one thing I am not sure is where to put `desired_secret_key`.

**Comment by marshtompsxd** [comment]
> @marshtompsxd , one thing I am not sure is where to put `desired_secret_key`.

If it is used by the liveness theorem, then it should go to trusted

**Comment by marshtompsxd** [comment]
Is this ready for review?

**Comment by euclidgame** [comment]
Yes

File: src/controller_examples/fluent_controller/fluentbit/trusted/liveness_theorem.rs [hunk 1]
```diff
@@ -1,10 +1,7 @@
...
 // SPDX-License-Identifier: MIT
 #![allow(unused_imports)]
-use crate::fluent_controller::fluentbit::{
-    common::*,
-    spec::{reconciler::*, resource::*, types::*},
-};
+use crate::fluent_controller::fluentbit::trusted::{maker::*, spec_types::*, step::*};
 use crate::kubernetes_api_objects::prelude::*;
 use crate::kubernetes_cluster::spec::{cluster::*, cluster_state_machine::Step, message::*};
 use crate::temporal_logic::{defs::*, rules::*};
```
Comment (marshtompsxd) [inline src/controller_examples/fluent_controller/fluentbit/trusted/liveness_theorem.rs:7]: Can you remove the `rules` here?

File: src/controller_examples/fluent_controller/fluentbit_config/trusted/liveness_theorem.rs [hunk 2]
```diff
@@ -1,10 +1,7 @@
...
 // SPDX-License-Identifier: MIT
 #![allow(unused_imports)]
-use crate::fluent_controller::fluentbit_config::{
-    common::*,
-    spec::{reconciler::*, resource::*, types::*},
-};
+use crate::fluent_controller::fluentbit_config::trusted::{maker::*, spec_types::*, step::*};
 use crate::kubernetes_api_objects::prelude::*;
 use crate::kubernetes_cluster::spec::{cluster::*, cluster_state_machine::Step, message::*};
 use crate::temporal_logic::{defs::*, rules::*};
```
Comment (marshtompsxd) [inline src/controller_examples/fluent_controller/fluentbit_config/trusted/liveness_theorem.rs:7]: Can you remove the `rules` here?

File: src/controller_examples/zookeeper_controller/trusted/spec_types.rs [hunk 3]
```diff
@@ -7,7 +7,7 @@ use crate::kubernetes_api_objects::{
 };
 use crate::kubernetes_cluster::spec::{cluster::*, cluster_state_machine::*, message::*};
 use crate::vstd_ext::string_view::*;
-use crate::zookeeper_controller::{common::*, spec::zookeeper_api::*};
+use crate::zookeeper_controller::trusted::{step::*, zookeeper_api_model::*};
```
Comment (marshtompsxd) [inline src/controller_examples/zookeeper_controller/trusted/spec_types.rs:10]: Maybe I missed anything, but does this file need to import zookeeper_api_model?

File: src/controller_examples/zookeeper_controller/trusted/zookeeper_api_model.rs
Comment (marshtompsxd) [inline src/controller_examples/zookeeper_controller/trusted/zookeeper_api_model.rs:1]: Can you rename this one to `zookeeper_api_spec`? I think in our context we will use `model` to refer to the specification that the impl is proved to conform to.

File: src/controller_examples/zookeeper_controller/trusted/zookeeper_api_model.rs [hunk 4]
```diff
@@ -5,8 +5,8 @@ use crate::kubernetes_api_objects::{
     common::*, config_map::*, dynamic::*, resource::*, stateful_set::*,
 };
 use crate::vstd_ext::string_view::*;
-use crate::zookeeper_controller::common::*;
-use crate::zookeeper_controller::spec::{types::ZookeeperClusterView, resource::make_zk_config};
+use crate::zookeeper_controller::model::resource::make_zk_config;
```
Comment (marshtompsxd) [inline src/controller_examples/zookeeper_controller/trusted/zookeeper_api_model.rs:8]: The trusted code should not import from the model, so we should move the code to the trusted component.

File: src/controller_examples/zookeeper_controller/trusted/spec_types.rs (hunk 3 above)
Comment (euclidgame) [reply src/controller_examples/zookeeper_controller/trusted/spec_types.rs:10]:
```rust
pub type ZKStep = Step<ZKMessage>;

pub type ZKCluster = Cluster<ZookeeperClusterView, ZKAPI, ZookeeperReconciler>;

pub type ZKMessage = Message<ZKAPIInputView, ZKAPIOutputView>;
```

I think the `ZKAPI` and related types are define there?

Comment (marshtompsxd) [reply src/controller_examples/zookeeper_controller/trusted/spec_types.rs:10]: oh I see

File: src/controller_examples/zookeeper_controller/trusted/zookeeper_api_model.rs (hunk 4 above)
Comment (euclidgame) [reply src/controller_examples/zookeeper_controller/trusted/zookeeper_api_model.rs:8]: I think the structure of `trusted` is a bit rough. Should I just put in in zk_api_spec?

Comment (marshtompsxd) [reply src/controller_examples/zookeeper_controller/trusted/zookeeper_api_model.rs:8]: Can you create a new file `trusted/config_data_serialization.rs` and put it there?

Comment (marshtompsxd) [reply src/controller_examples/zookeeper_controller/trusted/zookeeper_api_model.rs:8]: Just edited --- let's just call it `configmap.rs`

**Pull request merged**